yt_dlp/extractor/weibo.py

```diff
@@ -1,134 +1,246 @@
             })
 
-        uploader = self._og_search_property(
-            'nick-name', webpage, 'uploader', default=None)
-
+    def _weibo_download_json(self, url, video_id, *args, fatal=True, note='Downloading JSON metadata', **kwargs):
+        webpage, urlh = self._download_webpage_handle(url, video_id, *args, fatal=fatal, note=note, **kwargs)
+        if urllib.parse.urlparse(urlh.url).netloc == 'passport.weibo.com':
+            self._update_visitor_cookies(video_id)
+            webpage = self._download_webpage(url, video_id, *args, fatal=fatal, note=note, **kwargs)
+        return self._parse_json(webpage, video_id, fatal=fatal)
+
+    def _extract_formats(self, video_info):
+        media_info = traverse_obj(video_info, ('page_info', 'media_info'))
+        formats = [{
+            **traverse_obj(play_info, {
+                'url': ('url', {url_or_none}),
+                'format': ('quality_desc', {str_or_none}),
+                'format_id': ('label', {str_or_none}),
+                'ext': ('mime', {mimetype2ext}),
+                'tbr': ('bitrate', {int_or_none}, {lambda i: i if i else None}),
+                'vcodec': ('video_codecs', {str_or_none}),
+                'fps': ('fps', {int_or_none}),
+                'width': ('width', {int_or_none}),
+                'height': ('height', {int_or_none}),
+                'filesize': ('size', {int_or_none}),
+                'acodec': ('audio_codecs', {str_or_none}),
+                'asr': ('audio_sample_rate', {int_or_none}),
+                'audio_channels': ('audio_channels', {int_or_none}),
+            })
+        } for play_info in traverse_obj(media_info, ('playback_list', ..., 'play_info'))]
+        formats = [i for i in formats if i.get('url')]
+        if not formats:  # fallback, should be barely used
+            for url in set(traverse_obj(media_info, (..., {url_or_none}))):
+                if 'label=' in url:
+                    format_id, resolution = self._search_regex(
+                        r'label=(\w+)&template=(\d+x\d+)', url, 'format info',
+                        group=(1, 2), default=(None, None))
+                    formats.append({
+                        'url': url,
+                        'format_id': format_id,
+                        **parse_resolution(resolution),
+                        **traverse_obj(media_info, (
+                            'video_details', lambda _, i: i.get('label', '').startswith(format_id), {
+                                'size': ('size', {int_or_none}),
+                                'tbr': ('bitrate', {int_or_none}),
+                            }
+                        ), get_all=False),
+                    })
+        return formats
+
+    def _parse_video_info(self, video_info, video_id=None):
         return {
             'id': video_id,
-            'title': title,
-            'uploader': uploader,
-            'formats': formats
+            'formats': self._extract_formats(video_info),
```

Comment:
We should also do this, since the `UserIE` is also using this, so that downloads are recorded under `WeiboIE` in the download archive
```suggestion
            'extractor_key': WeiboIE.ie_key(),
            'extractor': WeiboIE.IE_NAME,
            'formats': self._extract_formats(video_info),
```

Reply:
Should we also add class variable for IE_NAME?

Reply:
Only need to define it if you want it to be something other than the class name minus `IE` suffix